rancher2_bootstrap is failing on first tf run

Hi 

terrafrom 0.13.5

I'm using the docker provisoner to setup rancher, bootstrap and create a custom cluster.
```
# ------------------------------------------------------------------------------
# CREATE A RANDOM PASSWORD
# ------------------------------------------------------------------------------

resource "random_password" "password" {
  length           = 10
  special          = true
  override_special = "_%@"
}

# ------------------------------------------------------------------------------
# LOCAL VARIABLES
# ------------------------------------------------------------------------------

locals {
  rancher_console_pw = random_password.password.result
  env                = terraform.workspace
}

# ------------------------------------------------------------------------------
# SET PROVIDERS
# ------------------------------------------------------------------------------

provider "docker" {
  host = "ssh://${var.rancher_host_ssh}:22"
}

provider "rancher2" {
  alias     = "bootstrap"
  api_url   = "https://${var.rancher_api_url}"
  bootstrap = true
  insecure  = true
  retries   = 100
}

resource "rancher2_bootstrap" "admin" {
  depends_on = [module.docker_rancher]
  provider   = rancher2.bootstrap

  password  = local.rancher_console_pw
  telemetry = false
}

provider rancher2 {
  alias = "admin"

  api_url   = rancher2_bootstrap.admin.url
  token_key = rancher2_bootstrap.admin.token
  insecure  = true
}
```

This seems to fail on the first run, but is good on the second run. 

```
Initializing provider plugins...
- Using previously-installed terraform-providers/docker v2.7.2
- Using previously-installed hashicorp/null v3.0.0
- Using previously-installed hashicorp/local v2.0.0
- Using previously-installed rancher/rancher2 v1.10.4
- Using previously-installed hashicorp/random v3.0.0
- Using previously-installed hashicorp/kubernetes v1.13.3

random_password.password: Creating...
random_password.password: Creation complete after 0s [id=none]
module.docker_rancher.docker_image.image: Creating...
module.docker_rancher.docker_image.image: Still creating... [10s elapsed]
module.docker_rancher.docker_image.image: Still creating... [20s elapsed]
module.docker_rancher.docker_image.image: Still creating... [30s elapsed]
module.docker_rancher.docker_image.image: Still creating... [40s elapsed]
module.docker_rancher.docker_image.image: Creation complete after 45s [id=sha256:2159e92d1f8d90cce01eb0b7e533023e9c0b2d77dd366030af84741009afb374rancher/rancher:v2.5.1]
module.docker_rancher.docker_container.container: Creating...
module.docker_rancher.docker_container.container: Creation complete after 2s [id=f21b785556a0b1106d127031ee775ace2c680f1dc68ea278bdf93a25f6109162]
rancher2_bootstrap.admin: Creating...
rancher2_bootstrap.admin: Still creating... [10s elapsed]
rancher2_bootstrap.admin: Still creating... [20s elapsed]
rancher2_bootstrap.admin: Still creating... [30s elapsed]
rancher2_bootstrap.admin: Still creating... [40s elapsed]
rancher2_bootstrap.admin: Still creating... [50s elapsed]
rancher2_bootstrap.admin: Still creating... [1m0s elapsed]

Error: [ERROR] Connecting with user/pass: Bad response statusCode [403]. Status [403 Forbidden]. Body: [baseType=error, code=Forbidden, message=settings.management.cattle.io "k8s-version" is forbidden: User "user-r5tbb" cannot get resource "settings" in API group "management.cattle.io" at the cluster scope] from [https://sbox-k8s:9443/v3/settings/k8s-version]

  on main.tf line 36, in resource "rancher2_bootstrap" "admin":
  36: resource "rancher2_bootstrap" "admin" {
```

I'm stuck on this at the moment. If I run it the second time it works. 
Any advise pls ? 
Thanks

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 1.10.5 (Unreleased)
+## 1.10.5 (November 6, 2020)
 
 FEATURES:
 
@@ -7,11 +7,20 @@ FEATURES:
 
 ENHANCEMENTS:
 
-
+* Updated go mod to support Rancher `v2.5.2-rc2`
+* Updated acceptance tests to use Rancher `v2.5.2-rc2`
+* Improved `rancher2_bootstrap` on resource creation. Number of retires on `bootstrapDoLogin` function can be configured with `retries` provider argument
+* Updated `rancher2_catalog_v2` contextualized resource id with `cluster_id` prefix
+* Updated `rancher2_app_v2` contextualized resource id with `cluster_id` prefix
+* Updated `rancher2_app_v2` to show helm operation log if fail
+* Updated `rancher2_app_v2.values` argument as sensitive
 
 BUG FIXES:
 
-
+* Fixed `rancher2_cluster.rke_config.upgrade_strategy.drain` argument to set false value properly
+* Fixed `Apps & marketplace` guide for Rancher v2.5.0 format
+* Fixed `rancher2_app_v2.values` argument to avoid false diff
+* Fixed `rancher2_cluster_role_template_binding` and  `rancher2_cluster_role_template_binding` arguments to forceNew on update  
 
 ## 1.10.4 (October 29, 2020)
 
@@ -45,6 +54,7 @@ BUG FIXES:
 
 * Fix `rke_config.monitoring.replicas` argument to set default value to 1 if monitoring enabled
 * Fix Rancher auth config apply on activedirectory, freeipa and openldap providers
+* Fix `rancher2_cluster.rke_config.upgrade_strategy.drain` argument to set false value properly
 
 
 ## 1.10.3 (September 14, 2020)

--- a/docs/guides/apps_marketplace.md
+++ b/docs/guides/apps_marketplace.md
@@ -127,7 +127,8 @@ EOF
 }
 ```
 
-** tip ** If you are reinstalling `rancher-monitoring` and the deployment is failing, try adding this values to app v2
+**Tip:** If you are reinstalling `rancher-monitoring` and the deployment is failing, try adding this values to app v2
+
 ```
 alertmanager:
   alertmanagerSpec:
@@ -147,6 +148,11 @@ resource "rancher2_app_v2" "rancher-monitoring" {
   chart_name = "rancher-monitoring"
   chart_version = "9.4.200"
   values = <<EOF
+alertmanager:
+  alertmanagerSpec:
+    enabled: false
+    useExistingSecret: true
+    configSecret: alertmanager-rancher-monitoring-alertmanager
 prometheus:
   prometheusSpec:
     requests:

--- a/docs/resources/app_v2.md
+++ b/docs/resources/app_v2.md
@@ -32,7 +32,7 @@ The following arguments are supported:
 * `chart_name` - (Required) The app v2 chart name (string)
 * `chart_version` - (Optional) The app v2 chart version (string)
 * `project_id` - (Optional) Deploy the app v2 within project ID (string)
-* `values` - (Optional) The app v2 values yaml. Yaml format is required (string)
+* `values` - (Optional/Sensitive) The app v2 values yaml. Yaml format is required (string)
 * `cleanup_on_fail` - (Optional) Cleanup app v2 on failed chart upgrade. Default: `false` (bool)
 * `disable_hooks` - (Optional) Disable app v2 chart hooks. Default: `false` (bool)
 * `disable_open_api_validation` - (Optional) Disable app V2 Open API Validation. Default: `false` (bool)

--- a/go.mod
+++ b/go.mod
@@ -7,25 +7,24 @@ require (
 	github.com/hashicorp/go-version v1.2.0
 	github.com/hashicorp/terraform-plugin-sdk v1.14.0
 	github.com/rancher/client-go v11.0.0+incompatible // indirect
-	github.com/rancher/eks-operator v1.0.1 // indirect
 	github.com/rancher/lasso v0.0.0-20200905045615-7fcb07d6a20b // indirect
 	github.com/rancher/norman v0.0.0-20200930000340-693d65aaffe3
-	github.com/rancher/rancher v0.0.0-20201007163458-eb990af66be1
+	github.com/rancher/rancher v0.0.0-20201109223515-3205bc1e82e5
 	github.com/rancher/rancher/pkg/apis v0.0.0
 	github.com/rancher/rancher/pkg/client v0.0.0
-	github.com/rancher/wrangler v0.7.2
+	github.com/rancher/wrangler v0.7.3-0.20201023210123-9b7d265b90ec // indirect
 	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
 	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/api v0.19.0
+	k8s.io/api v0.19.0 // indirect
 	k8s.io/apimachinery v0.19.0
 	k8s.io/apiserver v0.19.0
-	k8s.io/client-go v12.0.0+incompatible
+	k8s.io/client-go v12.0.0+incompatible // indirect
 )
 
 replace (
 	github.com/crewjam/saml => github.com/crewjam/saml v0.4.1
-	github.com/rancher/rancher/pkg/apis => github.com/rancher/rancher/pkg/apis v0.0.0-20201007163458-eb990af66be1
-	github.com/rancher/rancher/pkg/client => github.com/rancher/rancher/pkg/client v0.0.0-20201007163458-eb990af66be1
+	github.com/rancher/rancher/pkg/apis => github.com/rancher/rancher/pkg/apis v0.0.0-20201109223515-3205bc1e82e5
+	github.com/rancher/rancher/pkg/client => github.com/rancher/rancher/pkg/client v0.0.0-20201109223515-3205bc1e82e5
 	k8s.io/api => k8s.io/api v0.19.0
 	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.19.0
 	k8s.io/apimachinery => github.com/rancher/apimachinery v0.19.0-rancher1

--- a/rancher2/0_provider_upgrade_test.go
+++ b/rancher2/0_provider_upgrade_test.go
@@ -38,7 +38,7 @@ resource "rancher2_namespace" "testacc" {
   project_id = rancher2_cluster_sync.testacc.default_project_id
 }
 `
-	testAccCheckRancher2UpgradeVersion = []string{"v2.3.6", "v2.4.8", "v2.5.1"}
+	testAccCheckRancher2UpgradeVersion = []string{"v2.3.6", "v2.4.8", "v2.5.2"}
 	testAccCheckRancher2RunningVersionIndex = 0
 	testAccCheckRancher2UpgradeCluster = os.Getenv("RANCHER_ACC_CLUSTER_NAME")
 	testAccCheckRancher2UpgradeCatalogV24 = testAccRancher2CatalogGlobal + testAccRancher2CatalogCluster + testAccRancher2CatalogProject

--- a/rancher2/config.go
+++ b/rancher2/config.go
@@ -84,7 +84,7 @@ func (c *Config) isRancherReady() error {
 	var resp []byte
 	url := RootURL(c.URL) + "/ping"
 	for i := 0; i <= c.Retries; i++ {
-		resp, err = DoGet(url, "", "", c.CACerts, c.Insecure)
+		resp, err = DoGet(url, "", "", "", c.CACerts, c.Insecure)
 		if err == nil && rancher2ReadyAnswer == string(resp) {
 			return nil
 		}
@@ -758,6 +758,25 @@ func (c *Config) GetAppV2OperationByID(clusterID, id string) (map[string]interfa
 	err = client.ByID(appV2OperationAPIType, id, &resp)
 
 	return resp, err
+}
+
+func (c *Config) GetAppV2OperationLogs(clusterID string, op map[string]interface{}) (string, error) {
+	if op["id"].(string) == "" {
+		return "", fmt.Errorf("App V2 operation id is nil")
+	}
+
+	links := toMapString(op["links"].(map[string]interface{}))
+	link := "logs"
+	if links == nil && len(links[link]) == 0 {
+		return "", fmt.Errorf("failed to get app v2 operation log %s", op["id"])
+	}
+
+	resp, err := DoGet(links[link], "", "", c.TokenKey, c.CACerts, c.Insecure)
+	if err != nil {
+		return "", fmt.Errorf("failed to get app v2 operation log %s: %s", op["id"], err)
+	}
+
+	return string(resp), nil
 }
 
 func (c *Config) DeleteAppV2(clusterID string, app *AppV2) error {

--- a/rancher2/resource_rancher2_app_v2.go
+++ b/rancher2/resource_rancher2_app_v2.go
@@ -60,7 +60,7 @@ func resourceRancher2AppV2Create(d *schema.ResourceData, meta interface{}) error
 	if err != nil {
 		return fmt.Errorf("[ERROR] installing App V2: %s", err)
 	}
-	d.SetId(chartInstallAction.Namespace + "/" + name)
+	d.SetId(clusterID + appV2ClusterIDsep + chartInstallAction.Namespace + "/" + name)
 
 	return resourceRancher2AppV2Read(d, meta)
 }
@@ -77,8 +77,8 @@ func resourceRancher2AppV2Read(d *schema.ResourceData, meta interface{}) error {
 		}
 		d.Set("cluster_name", cluster.Name)
 	}
-
-	app, err := meta.(*Config).GetAppV2ByID(clusterID, d.Id())
+	_, rancherID := splitID(d.Id())
+	app, err := meta.(*Config).GetAppV2ByID(clusterID, rancherID)
 	if err != nil {
 		if IsNotFound(err) || IsForbidden(err) {
 			log.Printf("[INFO] App V2 %s not found at %s", name, clusterID)
@@ -123,7 +123,8 @@ func resourceRancher2AppV2Delete(d *schema.ResourceData, meta interface{}) error
 	name := d.Get("name").(string)
 	log.Printf("[INFO] Deleting App V2 %s at %s", name, clusterID)
 
-	app, err := meta.(*Config).GetAppV2ByID(clusterID, d.Id())
+	_, rancherID := splitID(d.Id())
+	app, err := meta.(*Config).GetAppV2ByID(clusterID, rancherID)
 	if err != nil {
 		if IsNotFound(err) || IsForbidden(err) {
 			log.Printf("[INFO] App V2 %s not found at %s", name, clusterID)
@@ -148,20 +149,16 @@ func resourceRancher2AppV2Delete(d *schema.ResourceData, meta interface{}) error
 	if waitErr != nil {
 		return fmt.Errorf("[ERROR] waiting for app (%s) to be deleted: %s", app.ID, waitErr)
 	}
-	if app.Spec.Chart.Metadata != nil && app.Spec.Chart.Metadata.Annotations != nil && len(app.Spec.Chart.Metadata.Annotations) > 0 {
+	if app.Spec.Chart.Metadata != nil && app.Spec.Chart.Metadata.Annotations != nil && len(app.Spec.Chart.Metadata.Annotations) > 0 && len(app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/auto-install"]) > 0 {
 		namespace := d.Get("namespace").(string)
-		name := ""
 		if len(app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/namespace"]) > 0 {
 			namespace = app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/namespace"]
 		}
-		if len(app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/auto-install"]) > 0 {
-			chartAuto := splitBySep(app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/auto-install"], "=")
-			if len(chartAuto) != 2 {
-				return fmt.Errorf("bad format on chart annotation catalog.cattle.io/auto-install: %s", app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/auto-install"])
-			}
-			name = chartAuto[0]
+		chartAuto := splitBySep(app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/auto-install"], "=")
+		if len(chartAuto) != 2 {
+			return fmt.Errorf("bad format on chart annotation catalog.cattle.io/auto-install: %s", app.Spec.Chart.Metadata.Annotations["catalog.cattle.io/auto-install"])
 		}
-
+		name := chartAuto[0]
 		app, err = meta.(*Config).GetAppV2ByID(clusterID, namespace+"/"+name)
 		if err != nil {
 			if IsNotFound(err) || IsForbidden(err) {
@@ -185,6 +182,7 @@ func resourceRancher2AppV2Delete(d *schema.ResourceData, meta interface{}) error
 		if waitErr != nil {
 			return fmt.Errorf("[ERROR] waiting for app (%s) to be deleted: %s", app.ID, waitErr)
 		}
+
 	}
 	return nil
 }

--- a/rancher2/resource_rancher2_app_v2.go
+++ b/rancher2/resource_rancher2_app_v2.go
@@ -213,7 +213,11 @@ func appV2OperationWait(meta interface{}, clusterID, opID string) error {
 			if state, ok := metadata["state"].(map[string]interface{}); ok && len(state) > 0 {
 				if transitioning, ok := state["transitioning"].(bool); ok && !transitioning {
 					if opError, ok := state["error"].(bool); ok && opError {
-						return fmt.Errorf("%s", state["message"])
+						message, err := meta.(*Config).GetAppV2OperationLogs(clusterID, obj)
+						if err != nil {
+							return fmt.Errorf("%s: %s", state["message"], err)
+						}
+						return fmt.Errorf("%s", message)
 					}
 					return nil
 				}

--- a/rancher2/resource_rancher2_app_v2_test.go
+++ b/rancher2/resource_rancher2_app_v2_test.go
@@ -164,7 +164,8 @@ func testAccRancher2AppV2Disappears(cat *AppV2) resource.TestCheckFunc {
 
 			clusterID := rs.Primary.Attributes["cluster_id"]
 			name := rs.Primary.Attributes["name"]
-			app, err := testAccProvider.Meta().(*Config).GetAppV2ByID(clusterID, rs.Primary.ID)
+			_, rancherID := splitID(rs.Primary.ID)
+			app, err := testAccProvider.Meta().(*Config).GetAppV2ByID(clusterID, rancherID)
 			if err != nil {
 				if IsNotFound(err) || IsForbidden(err) {
 					return nil
@@ -206,7 +207,8 @@ func testAccCheckRancher2AppV2Exists(n string, cat *AppV2) resource.TestCheckFun
 		}
 
 		clusterID := rs.Primary.Attributes["cluster_id"]
-		foundReg, err := testAccProvider.Meta().(*Config).GetAppV2ByID(clusterID, rs.Primary.ID)
+		_, rancherID := splitID(rs.Primary.ID)
+		foundReg, err := testAccProvider.Meta().(*Config).GetAppV2ByID(clusterID, rancherID)
 		if err != nil {
 			if IsNotFound(err) || IsForbidden(err) {
 				return nil
@@ -226,7 +228,8 @@ func testAccCheckRancher2AppV2Destroy(s *terraform.State) error {
 			continue
 		}
 		clusterID := rs.Primary.Attributes["cluster_id"]
-		_, err := testAccProvider.Meta().(*Config).GetAppV2ByID(clusterID, rs.Primary.ID)
+		_, rancherID := splitID(rs.Primary.ID)
+		_, err := testAccProvider.Meta().(*Config).GetAppV2ByID(clusterID, rancherID)
 		if err != nil {
 			if IsNotFound(err) {
 				return nil

--- a/rancher2/resource_rancher2_bootstrap.go
+++ b/rancher2/resource_rancher2_bootstrap.go
@@ -259,8 +259,8 @@ func bootstrapDoLogin(d *schema.ResourceData, meta interface{}) error {
 	}
 	tokenID, token, err := DoUserLogin(meta.(*Config).URL, bootstrapDefaultUser, currentPass, bootstrapDefaultTTL, bootstrapDefaultSessionDesc, meta.(*Config).CACerts, meta.(*Config).Insecure)
 	if err != nil {
-		// 3 login retries waiting 5 seconds if user/pass login fails
-		for i := 0; i < 3; i++ {
+		// login retries waiting 5 seconds if user/pass login fails
+		for i := 0; i < meta.(*Config).Retries; i++ {
 			time.Sleep(5 * time.Second)
 			tokenID, token, err = DoUserLogin(meta.(*Config).URL, bootstrapDefaultUser, currentPass, bootstrapDefaultTTL, bootstrapDefaultSessionDesc, meta.(*Config).CACerts, meta.(*Config).Insecure)
 			if err == nil {

--- a/rancher2/resource_rancher2_catalog_v2.go
+++ b/rancher2/resource_rancher2_catalog_v2.go
@@ -39,19 +39,18 @@ func resourceRancher2CatalogV2Create(d *schema.ResourceData, meta interface{}) e
 	if err != nil {
 		return err
 	}
-	id := newCatalog.ID
-	d.SetId(id)
+	d.SetId(clusterID + catalogV2ClusterIDsep + newCatalog.ID)
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{},
 		Target:     []string{"active"},
-		Refresh:    catalogV2StateRefreshFunc(meta, clusterID, id),
+		Refresh:    catalogV2StateRefreshFunc(meta, clusterID, newCatalog.ID),
 		Timeout:    d.Timeout(schema.TimeoutCreate),
 		Delay:      1 * time.Second,
 		MinTimeout: 3 * time.Second,
 	}
 	_, waitErr := stateConf.WaitForState()
 	if waitErr != nil {
-		return fmt.Errorf("[ERROR] waiting for catalog (%s) to be active: %s", id, waitErr)
+		return fmt.Errorf("[ERROR] waiting for catalog (%s) to be active: %s", newCatalog.ID, waitErr)
 	}
 	return resourceRancher2CatalogV2Read(d, meta)
 }
@@ -61,7 +60,8 @@ func resourceRancher2CatalogV2Read(d *schema.ResourceData, meta interface{}) err
 	name := d.Get("name").(string)
 	log.Printf("[INFO] Refreshing Catalog V2 %s", name)
 
-	catalog, err := meta.(*Config).GetCatalogV2ByID(clusterID, d.Id())
+	_, rancherID := splitID(d.Id())
+	catalog, err := meta.(*Config).GetCatalogV2ByID(clusterID, rancherID)
 	if err != nil {
 		if IsNotFound(err) || IsForbidden(err) {
 			log.Printf("[INFO] Catalog V2 %s not found", name)
@@ -79,11 +79,12 @@ func resourceRancher2CatalogV2Update(d *schema.ResourceData, meta interface{}) e
 	catalog := expandCatalogV2(d)
 	log.Printf("[INFO] Updating Catalog V2 %s", name)
 
-	newCatalog, err := meta.(*Config).UpdateCatalogV2(clusterID, d.Id(), catalog)
+	_, rancherID := splitID(d.Id())
+	newCatalog, err := meta.(*Config).UpdateCatalogV2(clusterID, rancherID, catalog)
 	if err != nil {
 		return err
 	}
-	d.SetId(newCatalog.ID)
+	d.SetId(clusterID + catalogV2ClusterIDsep + newCatalog.ID)
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{},
 		Target:     []string{"active"},
@@ -104,7 +105,8 @@ func resourceRancher2CatalogV2Delete(d *schema.ResourceData, meta interface{}) e
 	name := d.Get("name").(string)
 	log.Printf("[INFO] Deleting Catalog V2 %s", name)
 
-	catalog, err := meta.(*Config).GetCatalogV2ByID(clusterID, d.Id())
+	_, rancherID := splitID(d.Id())
+	catalog, err := meta.(*Config).GetCatalogV2ByID(clusterID, rancherID)
 	if err != nil {
 		if IsNotFound(err) || IsForbidden(err) {
 			d.SetId("")

--- a/rancher2/resource_rancher2_catalog_v2_test.go
+++ b/rancher2/resource_rancher2_catalog_v2_test.go
@@ -105,7 +105,8 @@ func testAccRancher2CatalogV2Disappears(cat *ClusterRepo) resource.TestCheckFunc
 				continue
 			}
 			clusterID := rs.Primary.Attributes["cluster_id"]
-			catalog, err := testAccProvider.Meta().(*Config).GetCatalogV2ByID(clusterID, rs.Primary.ID)
+			_, rancherID := splitID(rs.Primary.ID)
+			catalog, err := testAccProvider.Meta().(*Config).GetCatalogV2ByID(clusterID, rancherID)
 			if err != nil {
 				if IsNotFound(err) || IsForbidden(err) {
 					return nil
@@ -147,7 +148,8 @@ func testAccCheckRancher2CatalogV2Exists(n string, cat *ClusterRepo) resource.Te
 		}
 
 		clusterID := rs.Primary.Attributes["cluster_id"]
-		foundReg, err := testAccProvider.Meta().(*Config).GetCatalogV2ByID(clusterID, rs.Primary.ID)
+		_, rancherID := splitID(rs.Primary.ID)
+		foundReg, err := testAccProvider.Meta().(*Config).GetCatalogV2ByID(clusterID, rancherID)
 		if err != nil {
 			if IsNotFound(err) || IsForbidden(err) {
 				return nil
@@ -167,7 +169,8 @@ func testAccCheckRancher2CatalogV2Destroy(s *terraform.State) error {
 			continue
 		}
 		clusterID := rs.Primary.Attributes["cluster_id"]
-		_, err := testAccProvider.Meta().(*Config).GetCatalogV2ByID(clusterID, rs.Primary.ID)
+		_, rancherID := splitID(rs.Primary.ID)
+		_, err := testAccProvider.Meta().(*Config).GetCatalogV2ByID(clusterID, rancherID)
 		if err != nil {
 			if IsNotFound(err) {
 				return nil

--- a/rancher2/resource_rancher2_cluster_test.go
+++ b/rancher2/resource_rancher2_cluster_test.go
@@ -107,7 +107,7 @@ resource "` + testAccRancher2ClusterType + `" "foo" {
       }
     }
     upgrade_strategy {
-      drain = true
+      drain = false
       max_unavailable_worker = "10%"
     }
   }
@@ -209,7 +209,7 @@ func TestAccRancher2Cluster_basic_RKE(t *testing.T) {
 					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "rke_config.0.services.0.etcd.0.creation", "12h"),
 					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "rke_config.0.services.0.etcd.0.retention", "72h"),
 					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "rke_config.0.services.0.kube_api.0.audit_log.0.configuration.0.max_age", "7"),
-					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "rke_config.0.upgrade_strategy.0.drain", "true"),
+					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "rke_config.0.upgrade_strategy.0.drain", "false"),
 					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "rke_config.0.upgrade_strategy.0.max_unavailable_worker", "10%"),
 					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "scheduled_cluster_scan.0.scan_config.0.cis_scan_config.0.debug_worker", "true"),
 					resource.TestCheckResourceAttr(testAccRancher2ClusterType+".foo", "scheduled_cluster_scan.0.schedule_config.0.cron_schedule", "30 10 * * *"),

--- a/rancher2/schema_app_v2.go
+++ b/rancher2/schema_app_v2.go
@@ -79,6 +79,7 @@ func appV2Fields() map[string]*schema.Schema {
 		"values": {
 			Type:        schema.TypeString,
 			Optional:    true,
+			Sensitive:   false,
 			Description: "App v2 custom values yaml",
 			ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
 				v, ok := val.(string)
@@ -102,7 +103,7 @@ func appV2Fields() map[string]*schema.Schema {
 				globalInfo := map[string]interface{}{
 					"cattle": map[string]interface{}{
 						"clusterId":   d.Get("cluster_id").(string),
-						"clusterName": d.Get("cluster_id").(string),
+						"clusterName": d.Get("cluster_name").(string),
 					},
 				}
 				if newGlobal, ok := newMap["global"].(map[string]interface{}); ok && len(newGlobal) > 0 {

--- a/rancher2/schema_app_v2.go
+++ b/rancher2/schema_app_v2.go
@@ -16,6 +16,7 @@ const (
 	appV2APIType          = rancher2CatalogTypePrefix + ".app"
 	appV2OperationAPIType = rancher2CatalogTypePrefix + ".operation"
 	appV2ValueGlobal      = "global."
+	appV2ClusterIDsep     = "."
 )
 
 //Types

--- a/rancher2/schema_catalog_v2.go
+++ b/rancher2/schema_catalog_v2.go
@@ -7,10 +7,11 @@ import (
 )
 
 const (
-	catalogV2Kind       = "ClusterRepo"
-	catalogV2APIGroup   = "catalog.cattle.io"
-	catalogV2APIVersion = "v1"
-	catalogV2APIType    = rancher2CatalogTypePrefix + ".clusterrepo"
+	catalogV2Kind         = "ClusterRepo"
+	catalogV2APIGroup     = "catalog.cattle.io"
+	catalogV2APIVersion   = "v1"
+	catalogV2APIType      = rancher2CatalogTypePrefix + ".clusterrepo"
+	catalogV2ClusterIDsep = "."
 )
 
 //Types

--- a/rancher2/structure_app_v2.go
+++ b/rancher2/structure_app_v2.go
@@ -16,9 +16,6 @@ func flattenAppV2(d *schema.ResourceData, in *AppV2) error {
 		return nil
 	}
 
-	if len(in.ID) > 0 {
-		d.SetId(in.ID)
-	}
 	d.Set("name", in.ObjectMeta.Name)
 	d.Set("namespace", in.ObjectMeta.Namespace)
 	err := d.Set("annotations", toMapInterface(in.ObjectMeta.Annotations))
@@ -46,8 +43,14 @@ func flattenAppV2(d *schema.ResourceData, in *AppV2) error {
 				if clusterID, ok := cattle["clusterId"].(string); ok && len(clusterID) > 0 {
 					d.Set("cluster_id", clusterID)
 				}
+				if clusterName, ok := cattle["clusterName"].(string); ok && len(clusterName) > 0 {
+					d.Set("cluster_name", clusterName)
+				}
 			}
 		}
+	}
+	if len(in.ID) > 0 {
+		d.SetId(d.Get("cluster_id").(string) + appV2ClusterIDsep + in.ID)
 	}
 
 	return nil

--- a/rancher2/structure_catalog_v2.go
+++ b/rancher2/structure_catalog_v2.go
@@ -13,7 +13,7 @@ func flattenCatalogV2(d *schema.ResourceData, in *ClusterRepo) error {
 	}
 
 	if len(in.ID) > 0 {
-		d.SetId(in.ID)
+		d.SetId(d.Get("cluster_id").(string) + catalogV2ClusterIDsep + in.ID)
 	}
 	d.Set("name", in.ObjectMeta.Name)
 	err := d.Set("annotations", toMapInterface(in.ObjectMeta.Annotations))
@@ -53,7 +53,7 @@ func expandCatalogV2(in *schema.ResourceData) *ClusterRepo {
 	obj := &ClusterRepo{}
 
 	if len(in.Id()) > 0 {
-		obj.ID = in.Id()
+		_, obj.ID = splitID(in.Id())
 	}
 	obj.TypeMeta.Kind = catalogV2Kind
 	obj.TypeMeta.APIVersion = catalogV2APIGroup + "/" + catalogV2APIVersion

--- a/rancher2/structure_cluster_rke_config_nodes.go
+++ b/rancher2/structure_cluster_rke_config_nodes.go
@@ -29,7 +29,9 @@ func flattenClusterRKEConfigNodeUpgradeStrategy(in *managementClient.NodeUpgrade
 		return []interface{}{}
 	}
 
-	obj["drain"] = in.Drain
+	if in.Drain != nil {
+		obj["drain"] = *in.Drain
+	}
 
 	if in.DrainInput != nil {
 		obj["drain_input"] = flattenClusterRKEConfigNodeDrainInput(in.DrainInput)
@@ -144,7 +146,7 @@ func expandClusterRKEConfigNodeUpgradeStrategy(p []interface{}) *managementClien
 	in := p[0].(map[string]interface{})
 
 	if v, ok := in["drain"].(bool); ok {
-		obj.Drain = v
+		obj.Drain = &v
 	}
 
 	if v, ok := in["drain_input"].([]interface{}); ok {

--- a/rancher2/structure_cluster_rke_config_nodes_test.go
+++ b/rancher2/structure_cluster_rke_config_nodes_test.go
@@ -34,7 +34,7 @@ func init() {
 		},
 	}
 	testClusterRKEConfigNodeUpgradeStrategyConf = &managementClient.NodeUpgradeStrategy{
-		Drain:                      false,
+		Drain:                      newFalse(),
 		DrainInput:                 testClusterRKEConfigNodeDrainInputConf,
 		MaxUnavailableControlplane: "2",
 		MaxUnavailableWorker:       "20%",

--- a/rancher2/util.go
+++ b/rancher2/util.go
@@ -177,7 +177,7 @@ func DoPost(url, data, cacert string, insecure bool, headers map[string]string) 
 	return response, nil
 }
 
-func DoGet(url, username, password, cacert string, insecure bool) ([]byte, error) {
+func DoGet(url, username, password, token, cacert string, insecure bool) ([]byte, error) {
 	start := time.Now()
 
 	if url == "" {
@@ -186,13 +186,16 @@ func DoGet(url, username, password, cacert string, insecure bool) ([]byte, error
 	log.Println("Getting from ", url)
 
 	client := &http.Client{
-		Timeout: time.Duration(10 * time.Second),
+		Timeout: time.Duration(60 * time.Second),
 		CheckRedirect: func(req *http.Request, via []*http.Request) error {
 			if len(via) >= maxHTTPRedirect {
 				return fmt.Errorf("Stopped after %d redirects", maxHTTPRedirect)
 			}
-			if len(username) > 0 && len(password) > 0 {
-				req.SetBasicAuth(username, password)
+			if len(token) > 0 {
+				req.Header.Add("Authorization", "Bearer "+token)
+			} else if len(username) > 0 && len(password) > 0 {
+				s := username + ":" + password
+				req.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(s)))
 			}
 			return nil
 		},
@@ -222,8 +225,11 @@ func DoGet(url, username, password, cacert string, insecure bool) ([]byte, error
 	if err != nil {
 		return nil, fmt.Errorf("Doing get: %v", err)
 	}
-	if len(username) > 0 && len(password) > 0 {
-		req.SetBasicAuth(username, password)
+	if len(token) > 0 {
+		req.Header.Add("Authorization", "Bearer "+token)
+	} else if len(username) > 0 && len(password) > 0 {
+		s := username + ":" + password
+		req.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(s)))
 	}
 	resp, err := client.Do(req)
 	if err != nil {

--- a/scripts/start_rancher.sh
+++ b/scripts/start_rancher.sh
@@ -20,7 +20,7 @@ CERTMANAGER_CRD=${CERTMANAGER_CRD:-"https://github.com/jetstack/cert-manager/rel
 CERTMANAGER_NS=${CERTMANAGER_NS:-"cert-manager"}
 
 ## rancher
-RANCHER_VERSION=${RANCHER_VERSION:-"v2.5.1"}
+RANCHER_VERSION=${RANCHER_VERSION:-"v2.5.2"}
 RANCHER_NS=${RANCHER_NS:-"cattle-system"}
 export RANCHER_HOSTNAME="rancher.${K3S_SERVER_IP}.xip.io"
 export RANCHER_IP=${K3S_SERVER_IP}

